spikeinterface/core/old_api_utils.py

```diff
@@ -51,6 +51,176 @@ def create_extractor_from_new_recording(new_recording):
+        and end_frame is exclusive conforming to numpy standards.
+
+        Parameters
+        ----------
+        unit_id: int
+            The id that specifies a unit in the recording
+        start_frame: int
+            The frame above which a spike frame is returned  (inclusive)
+        end_frame: int
+            The frame below which a spike frame is returned  (exclusive)
+
+        Returns
+        -------
+        spike_train: numpy.ndarray
+            An 1D array containing all the frames for each spike in the
+            specified unit given the range of start and end frames
+        """
+        return self._sorting.get_unit_spike_train(unit_id=self._unit_map[unit_id], segment_index=0,
+                                                  start_frame=start_frame, end_frame=end_frame)
+
+    def get_units_spike_train(self, unit_ids=None, start_frame=None, end_frame=None):
+        """This function extracts spike frames from the specified units.
+
+        Parameters
+        ----------
+        unit_ids: array_like
+            The unit ids from which to return spike trains. If None, all unit
+            spike trains will be returned
+        start_frame: int
+            The frame above which a spike frame is returned  (inclusive)
+        end_frame: int
+            The frame below which a spike frame is returned  (exclusive)
+
+        Returns
+        -------
+        spike_train: numpy.ndarray
+            An 2D array containing all the frames for each spike in the
+            specified units given the range of start and end frames
+        """
+        if unit_ids is None:
+            unit_ids = self.get_unit_ids()
+        spike_trains = [self.get_unit_spike_train(uid, start_frame, end_frame) for uid in unit_ids]
+        return spike_trains
+
+    def get_sampling_frequency(self):
+        """
+        It returns the sampling frequency.
+
+        Returns
+        -------
+        sampling_frequency: float
+            The sampling frequency
+        """
+        return self._sampling_frequency
+
+    def set_sampling_frequency(self, sampling_frequency):
+        """
+        It sets the sorting extractor sampling frequency.
+
+        Parameters
+        ----------
+        sampling_frequency: float
+            The sampling frequency
+        """
+        self._sampling_frequency = sampling_frequency
+
+    def set_times(self, times):
+        """This function sets the sorting times to convert spike trains to seconds
+
+        Parameters
+        ----------
+        times: array-like
+            The times in seconds for each frame
+        """
+        max_frames = np.array([np.max(self.get_unit_spike_train(u)) for u in self.get_unit_ids()])
+        assert np.all(max_frames < len(times)), "The length of 'times' should be greater than the maximum " \
+                                                "spike frame index"
+        self._times = times.astype('float64')
```

Comment:
Same here. times is not handle in the new API. I would raise an error.

Reply:
Unless we are OK that this times do not propagate to the new API, it is only something bound to this object.

Reply:
Done